### DISCUSSION TITLE
Mapping settings not saved

### DISCUSSION THREAD

**Discussion by WolfRamiO74** [darthcloud/BlueRetro#877] [Q&A] [answered]
Hello, I bought a "Blueretro Mini" retroscaler adapter for PS2 on Aliexpress... the issue is the following, it works perfectly except that some buttons are poorly mapped, 

I have connected from the PC to the adapter's web interface via Bluetooth and configure mapping so you i use an XBOX Oner S controller on the PS2. but when saving the changes the website says that they were saved but when checking if these were made the buttons are still incorrectly mapped.

I updated the firmware to 1.9.1
Any suggestions? thank you

**Comment by darthcloud** [comment]
Can you take screenshoot of what setting your are changing and were you are clicking to save.

No need to spam the whole place btw

**Reply by WolfRamiO74** [reply]
I found something, if I leave SRC and DST in "default" and make changes in the mapping table then press "save", the changes are saved, but if I make any changes using another element in SRC or DST that is not "default", changes are not saved.

**Comment by WolfRamiO74** [comment]
![Screenshotaaa_1](https://github.com/darthcloud/BlueRetro/assets/12602710/f93f2e39-a605-4cdf-a715-e13485a4fa42)


There... when i change SRC and DEST from default to those the click "Save" and reconnect to the device the changes were not applied,  the web show:

``
Config saved, mapping changes take effect immediately.

Live Output
Reading App version...
Reading Api version...
Requesting Bluetooth Device...
Connecting to GATT Server...
Getting BlueRetro Service...
ABI version: 2
Init Cfg DOM...
Get Global Config CHRC...
Reading Global Config...
Global Config size: 4
Get Output 0 CTRL CHRC...
Set Output 0 on CTRL chrc...
Get Output 0 DATA CHRC...
Reading Output 0 Config...
Output 0 Config size: 2
Get Input 0 Config CHRC...
Set Input Ctrl CHRC... 0
Reading Input Data CHRC...
Got Input Data 512
Got Input Data 97 97
Set Input Ctrl CHRC... 512
Reading Input Data CHRC...
Got Input Data 267
Got Input Data 97 0
Input 0 Config size: 2051
Input 0 Config size: 2051
Loading Mapping Found: 97 nbMapping: 97 cfg: 97
Set Input Ctrl CHRC... 0
Writing Input Data CHRC...
Input Data Written
Set Input Ctrl CHRC... 512
Writing Input Data CHRC...
Input Data Written
Input 0 Config saved
``

Thanks!

**Comment by darthcloud** [comment]
Can you screenshot the files listed on:
https://blueretro.io/files.html

Try delete config.bin?

**Reply by WolfRamiO74** [reply]
the changes were not applied
![Screenshotaaa_1](https://github.com/darthcloud/BlueRetro/assets/12602710/48f655b7-5c39-4111-b96a-9178b7069f2f)

OK, i deleted the config.bin file then try to makes the changes again, problem still persist, here is how default settings  form mapping looks like:

 
![Screenshotaaab_1](https://github.com/darthcloud/BlueRetro/assets/12602710/a3fc81d2-4d91-4f1e-a96e-86ce56c9b320)

**Comment by darthcloud** [comment, marked as answer]
Did you try to play a game and see if the mapping you did was there?

Also the mapping label are just an aid and doesn't save, but the actual settings does.

**Reply by WolfRamiO74** [reply]
you're right! I made the changes, saved and tested on the PS2, I don't know why it didn't work before, surely something happened in the testing process... 

I could be wrong but I think this "Also the mapping label are just an aid and doesn't save, but the actual settings does." is not in the wiki, if so it would be good if I would be. 

Thnaks!!!
